src/include/rbd/librbd.h

```diff
@@ -181,6 +181,22 @@ enum {
   RBD_IMAGE_OPTION_DATA_POOL = 10
 };
 
+typedef enum {
+  RBD_DELAY_CONDITION = 0,
+  RBD_CLONES_DELETED_CONDITION = 1
+} rbd_remove_condition_type_t;
+
+typedef struct {
+  uint64_t delay;
+} rbd_delay_condition_t;
+
+typedef struct {
+  rbd_remove_condition_type_t type;
+  union {
+    rbd_delay_condition_t delay_cond;
+  } cond;
+} rbd_remove_condition_t;
```

Comment:
To support this info, we will need to keep image entries in the RBD_TRASh object even after the image as been deleted.
Who will remove the entries for already purged images, and when?